### PR TITLE
Start process of dual licensing with 3-Clause BSD license

### PR DESCRIPTION
**Pull request by peterjc**
This is a work-in-progress pull request to address #898.

As it stands it only dual licenses a single file (_update:_ a few files), chosen because it unambiguously has approval from all contributors, for us to review the header language (and ideally have a lawyer look at it).

_Update: Proposed header:_

``` python
# This file is part of the Biopython distribution and governed by your
# choice of the "Biopython License Agreement" or the "BSD 3-Clause License".
# Please see the LICENSE file that should have been included as part of this
# package.
```

### CODE REVIEW

**Comment by peterjc** [comment]
The last commit adds a ``CONTRIBUTING`` file, see https://help.github.com/articles/setting-guidelines-for-repository-contributors/ (I'm assuming RST works here, if not we can switch it to MarkDown).

Perhaps we should also create ``.github/PULL_REQUEST_TEMPLATE.md`` with an explicit license agreement? See https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/

**Comment by peterjc** [comment]
Have I been consistent in spelling license with an s here? Over on #898 I often used licence with a c (UK spelling for the noun).

**Comment by peterjc** [comment]
The current wording in the ``LICENSE`` file header is too narrow when I talk about *Python files*, since we also have C source code. I was trying to avoid the more tricky issue about licensing for sample data. How about *source code* instead?

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+We welcome pull requests to fix bugs or add new features. Please read
+http://biopython.org/wiki/Contributing for a general overview about
+contributing - this file is primarily concerned with the practicalities.
+
+
+Licensing
+---------
+
+Biopython moving towards dual licensing the code. In yout git commit and/or
```
Comment (cbrueffer) [inline CONTRIBUTING]: moving -> is moving
yout -> your

File: CONTRIBUTING [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+
+Local Testing
+-------------
+
+Tools like ``pep8`` and ``flake8`` are useful, but currently we suggest you
+install this git pre-commit hook which will check our basic coding conventions
+as you work, https://github.com/biopython/biopython/issues/493
+
+Please allways run the full test suite on your local computer before
```
Comment (cbrueffer) [inline CONTRIBUTING]: allways -> always

File: CONTRIBUTING [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+submitting a pull request, e.g.::
+
+    $ git commit Bio/XXX.py Tests/test_XXX.py  -m "Fixed bug 123"
+    $ python3.5 setup.py build
+    $ python3.5 setup.py test
+
+If you have mutliple versions of Python installed, ideally test them all
```
Comment (cbrueffer) [inline CONTRIBUTING]: mutliple -> multiple

File: CONTRIBUTING [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Once you submit your pull request on GitHub, several automated tests should
+be run, and their results reported on the pull request.
+
+We use TravisCI to run most of the Biopython tests (although currently only
+under Linux, and not with all the optional dependencies included), plus also
+check the Python coding style using the ``pep8`` tool. This must pass before
+your pull request will be merged.
+
+We also are using QuantifiedCode which provides an independent code analysis
+which can catch important errors. Some of the issues it flags are subjecting,
```
Comment (cbrueffer) [inline CONTRIBUTING]: Not sure what's meant here, "subjective"?

**Comment by cbrueffer** [comment]
@peterjc Generally I like your "Python" -> "source code" suggestion; what about documentation though?

File: CONTRIBUTING (hunk 4 above)
Comment (peterjc) [inline CONTRIBUTING]: Yes - probably the Mac autocorrect did the best it could, and I didn't notice.

**Comment by peterjc** [comment]
@cbrueffer Good point, the Tutorial is also under the same licensing as the code. How about just "files" then, rather than "Python files"?

**Comment by cbrueffer** [comment]
@peterjc Yeah, that should work.

File: CONTRIBUTING [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Tools like ``pep8`` and ``flake8`` are very useful, but currently we suggest
+you install this git pre-commit hook which will check our basic coding
+conventions as you work, https://github.com/biopython/biopython/issues/493
+
+
+Testing
+-------
+
+Any new feature or functionality will not be accepted without tests. Likewise
+for any bug fix, we encourage an additional test is also included. See the
```
Comment (ericmjl) [inline CONTRIBUTING]: grammar fix, suggest: "we encourage including an additional test."

Comment (peterjc) [inline CONTRIBUTING]: Agreed, updating...

**Comment by twrightsman** [comment]
It seems the last paragraph of "coding conventions" and the first paragraph of "local testing" are almost the exact same paragraph. Is this intentionally redundant?

**Comment by peterjc** [comment]
@twrightsman well spotted - that duplication was a mistake, but which section do you think that text about pep8/flake8/pre-commit-hooks fits best?

**Comment by twrightsman** [comment]
I think "Coding Conventions" is the best spot for that information; a simple reminder to refer to the "Coding Conventions" section can be put at the end of testing if you believe its important that they keep the coding conventions in mind while reading the testing section as well.

**Comment by peterjc** [comment]
@twrightsman thanks, done.

**Comment by peterjc** [comment]
This needs something in the NEWS file, proposing:

```
We have started have begun to dual-license Biopython under both our original
liberal "Biopython License Agreement", and the very similar but more commonly
used "3-Clause BSD License". In this release a small number of the Python
files are explicitly available under either license, but most of the code
remains under the "Biopython License Agreement". See the LICENSE file for
more details.
```

**Pull request merged**